cmd/duffle/relocate.go

```diff
@@ -98,92 +106,78 @@ duffle relocate path/to/bundle.json --relocation-mapping path/to/relmap.json --r
 }
 
 func (r *relocateCmd) run() error {
-	bun, err := r.setup()
+	relMap := make(map[string]string)
+
+	rel, cleanup, err := r.setup()
 	if err != nil {
 		return err
 	}
+	defer cleanup()
+
+	if err := rel.Relocate(relMap); err != nil {
+		return err
+	}
 
-	return r.relocate(bun)
+	return r.writeRelocationMapping(relMap)
 }
 
-func (r *relocateCmd) relocate(bun *bundle.Bundle) error {
-	relMap := make(map[string]string)
-	for i := range bun.InvocationImages {
-		ii := bun.InvocationImages[i]
-		modified, err := r.relocateImage(&ii.BaseImage, relMap)
+// The caller is responsible for running the returned cleanup function, which may delete the returned bundle.
+func (r *relocateCmd) setup() (*relocator.Relocator, func(), error) {
+	nop := func() {}
+	dest := ""
+	bundleFile, err := resolveBundleFilePath(r.inputBundle, r.home.String(), r.bundleIsFile)
+	if err != nil {
+		return nil, nop, err
+	}
+
+	var bun *bundle.Bundle
+
+	if strings.HasSuffix(bundleFile, ".tgz") {
+		source, err := filepath.Abs(bundleFile)
```

Comment:
nit: I guess it would be more readable to extract this whole block its own sub function, like `func extractZippedBundle` ?
And what about changing this `if` with a switch case?
```go
switch{
case strings.HasSuffix(bundleFile, ".tgz"):
    dest, bun, err = extractZippedBundle()
    if err != nil{
         return nil, nop, err
    }
default:
    bun, err = loadBundle(bundleFile)
    if err != nil{
         return nil, nop, err
    }
}
```

Reply:
Extracted a function - good idea. I'll defer using a switch statement until there are other cases to consider.